backend/molgenis-emx2-semantics/src/main/java/org/molgenis/emx2/beaconv2/common/QueryHelper.java

```diff
@@ -73,8 +78,63 @@ public static OntologyTerm[] mapListToOntologyTerms(List<Map> mapList) {
    */
   public static OntologyTerm mapToOntologyTerm(Map map) {
     OntologyTerm ontologyTerm = new OntologyTerm();
+    if (map == null) {
+      return null;
+    }
     ontologyTerm.setId(map.get("codesystem") + ":" + TypeUtils.toString(map.get("code")));
     ontologyTerm.setLabel(TypeUtils.toString(map.get("name")));
     return ontologyTerm;
   }
+
+  /**
+   * Find column and GraphQL path to that column in a table based on a semantic tag.
+   *
+   * @param buildGraphQLFilter Supply empty string, will be filled by function recursively
+   * @param columnSemanticTagOrIRI Find column that matches this semantic tag
+   * @param table Table structure to search in
+   * @return
+   */
+  public static ColumnPath findColumnPath(
+      String buildGraphQLFilter, String columnSemanticTagOrIRI, Table table) {
+    for (Column column : table.getMetadata().getColumns()) {
+      if (column.getName().startsWith("mg_")) {
+        continue;
+      }
+
+      // check semantics, return if found
+      for (String semantics : column.getSemantics()) {
+        if (semantics.endsWith(columnSemanticTagOrIRI)) {
+          return new ColumnPath(column, buildGraphQLFilter + " {" + column.getName() + ": {");
+        }
+      }
+
+      // if reference, also step in recursively
+      if (column.isReference()) {
+        Table refTable = table.getSchema().getTable(column.getRefTableName());
+        // cannot refer to itself, circular
+        if (!refTable.getName().equals(table.getName())) {
```

Comment:
If you put the clause in a variable you don't need to put a comment any more: 
boolean isCircularReference = !refTable.getName().equals(table.getName());
if (isCircularReference) {......

You easily do these do this by selecting the clause and pressing ctrl-alt-v (check refactor context menu for the apple hotkey).

Reply:
No need to actually change this btw, just a tip for future code ;)

Reply:
good point, will actually update the code :)